### PR TITLE
Issue SB-4615 feat: Outline of meta filters added for review purpose only

### PR DESCRIPTION
**Pull request by revathitarento**
Outline function of meta filters added for review purpose only, PR is not for merge

### CODE REVIEW

**Comment by kochhar** [comment]
@revathip could you please fix the conflicts in this branch. second, please do not push to the release-1.7 branch

File: src/app.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ const learnerServiceLocalBaseUrl = process.env.sunbird_learner_service_local_bas
 
 const whiteListedChannelList = process.env.sunbird_content_service_whitelisted_channels
```
Comment (kochhar) [inline src/app.js]: now that metafilters are added, channel filters can be removed right?

File: src/app.js [hunk 2]
```diff
@@ -149,8 +152,8 @@ const telemetryConfig = {
 telemetry.init(telemetryConfig)
 
 // function to update the config
-function updateConfig (configString) {
-  configUtil.setConfig('CHANNEL_FILTER_QUERY_STRING', configString)
+function updateConfig (name, configString) {
```
Comment (kochhar) [inline src/app.js]: why is this function needed? can't the caller directly call configUtil.setConfig?

File: src/app.js [hunk 3]
```diff
@@ -171,3 +174,7 @@ function getFilterConfig () {
     filename, 'getFilterConfig', 'config string', configString))
   return configString
 }
+
+function getMetaFilterConfig () {
```
Comment (kochhar) [inline src/app.js]: I think this function could be dropped. Could getFilterConfig be used to get all the filter configurations?

File: src/routes/conceptRoutes.js [hunk 4]
```diff
@@ -27,7 +28,7 @@ module.exports = function (app) {
 
   app.route(BASE_URL_V1 + '/domains/:domainId/:objectType/search')
     .post(requestMiddleware.createAndValidateRequestBody, requestMiddleware.addChannelFilters,
-      domainService.searchObjectTypeAPI)
+      filterMiddleware.addMetaFilters, domainService.searchObjectTypeAPI)
```
Comment (kochhar) [inline src/routes/conceptRoutes.js]: channel filters should be merged in with metadata filters

File: src/routes/contentRoutes.js [hunk 5]
```diff
@@ -6,13 +6,14 @@
...
 var requestMiddleware = require('../middlewares/request.middleware')
+var filterMiddleware = require('../middlewares/filter.middleware')
 
 var BASE_URL = '/v1/content'
 
 module.exports = function (app) {
   app.route(BASE_URL + '/search')
     .post(requestMiddleware.createAndValidateRequestBody, requestMiddleware.addChannelFilters,
-      contentService.searchContentAPI)
+      filterMiddleware.addMetaFilters, contentService.searchContentAPI)
```
Comment (kochhar) [inline src/routes/contentRoutes.js]: same note as above on merging channel filters with metadata filters.

File: src/routes/frameworkCategoryInstanceRoutes.js [hunk 6]
```diff
@@ -15,7 +16,7 @@ module.exports = function (app) {
 
   app.route(baseUrl + '/search')
     .post(requestMiddleware.createAndValidateRequestBody, requestMiddleware.addChannelFilters,
-      frameworkCategoryInstanceService.frameworkCategoryInstanceSearch)
+      filterMiddleware.addMetaFilters, frameworkCategoryInstanceService.frameworkCategoryInstanceSearch)
```
Comment (kochhar) [inline src/routes/frameworkCategoryInstanceRoutes.js]: same note here.

File: src/routes/frameworkRoutes.js [hunk 7]
```diff
@@ -15,7 +16,7 @@ module.exports = function (app) {
 
   app.route(baseUrl + '/list')
     .post(requestMiddleware.createAndValidateRequestBody, requestMiddleware.addChannelFilters,
-      frameworkService.frameworklList)
+      filterMiddleware.addMetaFilters, frameworkService.frameworklList)
```
Comment (kochhar) [inline src/routes/frameworkRoutes.js]: same note here.

File: src/routes/frameworkTermRoutes.js [hunk 8]
```diff
@@ -15,7 +16,7 @@ module.exports = function (app) {
 
   app.route(baseUrl + '/search')
     .post(requestMiddleware.createAndValidateRequestBody, requestMiddleware.addChannelFilters,
-      frameworkTermService.frameworkTermSearch)
+      filterMiddleware.addMetaFilters, frameworkTermService.frameworkTermSearch)
```
Comment (kochhar) [inline src/routes/frameworkTermRoutes.js]: same note here.

File: src/routes/searchRoutes.js [hunk 9]
```diff
@@ -6,11 +6,12 @@
...
 var requestMiddleware = require('../middlewares/request.middleware')
+var filterMiddleware = require('../middlewares/filter.middleware')
 
 var BASE_URL_V1 = '/v1'
 
 module.exports = function (app) {
   app.route(BASE_URL_V1 + '/search')
     .post(requestMiddleware.createAndValidateRequestBody, requestMiddleware.addChannelFilters,
-      contentService.searchAPI)
+      filterMiddleware.addMetaFilters, contentService.searchAPI)
```
Comment (kochhar) [inline src/routes/searchRoutes.js]: same note here.

File: src/app.js (hunk 3 above)
Comment (kochhar) [reply src/app.js]: a skeleton implementation of the function which reads the metadata filter config values will be a good next step.

**Comment by gsbajaj72** [comment]
@revathip 
- Why have you taken latest from master branch and not from the release-1.8 branch.
- As discussed yesterday, the PR was supposed to be sent for release-1.8 branch
- Why is your local branch named release-1.7 and Not by feature name and JIRA ID

File: src/app.js (hunk 1 above)
Comment (revathitarento) [reply src/app.js]: Will be removed

File: src/app.js (hunk 2 above)
Comment (revathitarento) [reply src/app.js]: Done

File: src/routes/conceptRoutes.js (hunk 4 above)
Comment (revathitarento) [reply src/routes/conceptRoutes.js]: will be removed once i complete the meta filter implementation

File: src/routes/contentRoutes.js (hunk 5 above)
Comment (revathitarento) [reply src/routes/contentRoutes.js]: ok

File: src/routes/frameworkCategoryInstanceRoutes.js (hunk 6 above)
Comment (revathitarento) [reply src/routes/frameworkCategoryInstanceRoutes.js]: ok

File: src/routes/frameworkRoutes.js (hunk 7 above)
Comment (revathitarento) [reply src/routes/frameworkRoutes.js]: ok

**Comment by revathitarento** [comment]
Closing the PR, as it is raised to master branch

**Pull request closed without merging**